app/models/promotional_tweet.rb

```diff
@@ -13,4 +13,11 @@
 class PromotionalTweet < ActiveRecord::Base
   belongs_to :listen_signal
 
+  def update_posted_tweet_id(tweet_id)
+    update_attributes!(tweet_id: tweet_id, posted_at: Time.now.getutc)
+  end
+
+  def tweet_url(tweet_id)
```

Comment:
You shouldn't need to pass `tweet_id` in here. You can just reference it within the method.